qt-models/filtermodels.cpp

```diff
@@ -277,15 +277,12 @@ bool LocationFilterModel::doFilter(struct dive *d, QModelIndex &index0, QAbstrac
 			return true;
 	}
 
-	// there is a location selected
+	// There is a location selected
 	QStringList locationList = stringList();
-	if (!locationList.isEmpty()) {
-		locationList.removeLast(); // remove the "Show Empty Tags";
-		for (int i = 0; i < rowCount(); i++) {
-			if (checkState[i] && (location.indexOf(stringList()[i]) != -1)) {
-				return true;
-			}
-		}
+	// Ignore last item, since this is the "Show Empty Tags" entry
+	for (int i = 0; i < rowCount() - 1; i++) {
+		if (checkState[i] && location == locationList[i])
+			return true;
```

Comment:
I disagree with this change.
I want to see every dive that was at... what was that site called... "something wall"?
Type in "wall" and you see all of your dives at a site containing wall
I agree with you on Test 1 and Test 12 being annoying. So if you want to make this better
- ensure that the match is case insensitive (right now the code looks like it isn't)
- add an option to force exact match (that's your check for equality)
but by default it should match a substring

Reply:
Note that this code does *not* compare with user input. It compares strings stored in lists which were automatically generated (see the screenshot in #675). I don't think a substring search makes much sense in this case. Actually, this was the source of confusion that led to #675.

Of course, if the user entered the same dive site in different ways (whitespace, case, etc...) then this might be kind of useful, but I have my doubts that this is the correct place to catch this issue. And it still leaves the problem that the number of dives shown and the number of dives in parenthesis in the filter lists are inconsistent. I discovered that because I had the same principle idea as you: remove whitespace. It just ended up in a confusing user experience.

But at least now I understand where that code came from: It's probably a remnant of a time, when the user indeed could enter arbitrary strings.

Reply:
Yes, indeed, we allow arbitrary strings for filters - that's the text field above. But you are right, we then only filter against those that are checked.
I should stop trying to review your patches before having coffee. My apologies.